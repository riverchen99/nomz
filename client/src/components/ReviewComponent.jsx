import React from 'react';
import axios from 'axios';
import StarRatingComponent from 'react-star-rating-component';
import { StyledReviewComponent, UserInfo, UserAvatar, UserName, Rating, StarRating } from './StyledReviewComponent';
import avatar from '../images/default_profile.jpg'; // dummy avatar, will change later

class ReviewComponent extends React.Component {
  constructor(props) {
    super(props);
    
    this.state = {
      userName: '',
      starRating: 0,
      textReview: '',
    }
  }

  componentDidMount() {
    if (this.props.review != null) {
      axios.get(`/api/users/${this.props.review.author}`)
      .then((resp) => {
        this.setState({ userName: resp.data[0].name, starRating: this.props.review.rating, textReview: this.props.review.comments });
      })
      return 0;
    }
    else
      return 1;
  }

  render() {
    return (
      <StyledReviewComponent>
        <UserInfo>
          <UserAvatar src={avatar}/>
          <UserName>{this.state.userName}</UserName>
        </UserInfo>
        <Rating>
          <StarRating>
            <StarRatingComponent name="rating" editing={false} starCount={5} value={this.state.starRating} />
          </StarRating>
          {this.state.textReview}
        </Rating>
      </StyledReviewComponent>
    )
  }
}

export default ReviewComponent;
