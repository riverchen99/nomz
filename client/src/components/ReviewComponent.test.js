import React from 'react';
import { shallow, mount, configure } from 'enzyme';

import Adapter from 'enzyme-adapter-react-16';
import ReviewComponent from './ReviewComponent';

configure({ adapter: new Adapter() });

describe('Displaying review component', () => {
  it('should render correctly without props', () => {
    const component = shallow(<ReviewComponent />);
    expect(component).toMatchSnapshot();
  });
  it('should render correctly with given review prop', () => {
    const component = mount(<ReviewComponent review="5dcd9dada8d15f5897b49edb" />);
    expect(component.props().review).toEqual('5dcd9dada8d15f5897b49edb');
  });
});
