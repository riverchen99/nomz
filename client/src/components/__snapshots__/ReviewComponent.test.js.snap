// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Displaying review component should render correctly with given review prop 1`] = `ReactWrapper {}`;

exports[`Displaying review component should render correctly without props 1`] = `ShallowWrapper {}`;
